Error while checking build/O2/fullCI for 905d690eadf1dfb0a62af3823c2de19c608df5c7 at 2024-06-25 18:07:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
/sw/SOURCES/O2/13236-slc8_x86-64/0/Detectors/ITSMFT/common/reconstruction/include/ITSMFTReconstruction/RUDecodeData.h:146:38: error: no matching function for call to 'o2::itsmft::AlpideCoder::verifyDecodedCable(std::map<int, o2::itsmft::ChipPixelData*>&, std::array<o2::itsmft::PayLoadCont, 28>::value_type&, std::vector<short unsigned int>&, o2::itsmft::RUDecodeData::decodeROF<o2::itsmft::ChipMappingMFT>(const o2::itsmft::ChipMappingMFT&, o2::InteractionRecord, bool)::<lambda(int)>&)'
/sw/SOURCES/O2/13236-slc8_x86-64/0/Detectors/ITSMFT/common/reconstruction/include/ITSMFTReconstruction/RUDecodeData.h:146:38: error: no matching function for call to 'o2::itsmft::AlpideCoder::verifyDecodedCable(std::map<int, o2::itsmft::ChipPixelData*>&, std::array<o2::itsmft::PayLoadCont, 28>::value_type&, std::vector<short unsigned int>&, o2::itsmft::RUDecodeData::decodeROF<o2::itsmft::ChipMappingITS>(const o2::itsmft::ChipMappingITS&, o2::InteractionRecord, bool)::<lambda(int)>&)'
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/13236/905d690eadf1dfb0a62af3823c2de19c608df5c7/build_O2_fullCI/pretty.html).

--- a/Detectors/ITSMFT/common/reconstruction/include/ITSMFTReconstruction/AlpideCoder.h
+++ b/Detectors/ITSMFT/common/reconstruction/include/ITSMFTReconstruction/AlpideCoder.h
@@ -21,6 +21,7 @@
 #include "PayLoadCont.h"
 #include <map>
 #include <fmt/format.h>
+#include <iomanip>
 
 #include "ITSMFTReconstruction/PixelData.h"
 #include "ITSMFTReconstruction/DecodingStat.h"
@@ -105,6 +106,7 @@ class AlpideCoder
   static constexpr uint32_t DATASHORT = 0x4000; // flag for DATASHORT
   static constexpr uint32_t BUSYOFF = 0xf0;     // flag for BUSY_OFF
   static constexpr uint32_t BUSYON = 0xf1;      // flag for BUSY_ON
+  static constexpr uint32_t ERROR_MASK = 0xf0;  // flag for all error triggers
 
   // true if corresponds to DATALONG or DATASHORT: highest bit must be 0
   static bool isData(uint16_t v) { return (v & (0x1 << 15)) == 0; }
@@ -443,6 +445,154 @@ class AlpideCoder
       seenChips.push_back(chipData.getChipID());
     }
     return chipData.getData().size();
+  }
+
+  /// Verifies the decoder by comparing the contents a cable by re-encoding seen
+  /// chips back into the ALPIDE format.
+  template <typename LG, typename CG>
+  static bool verifyDecodedCable(
+    std::map<int, ChipPixelData*>& seenChips, PayLoadCont& buffer,
+    std::vector<uint16_t>& seenChipIDs, LG lidGetter, CG cidGetter)
+  {
+    PayLoadCont reconstructedData;
+
+    // Ensure the length of the reconstructed buffer.
+    int bufferLength = 0;
+    for (auto it = seenChips.begin(); it != seenChips.end(); ++it) {
+      bufferLength += it->second->getData().size();
+    }
+    bufferLength += seenChipIDs.size() * 2;
+    reconstructedData.ensureFreeCapacity(40 * bufferLength);
+
+    // Encode the seen chips in the order they were decoded.
+    for (int ID : seenChipIDs) {
+      ChipPixelData currentChip;
+      int localID = lidGetter(ID);
+      if (seenChips.count(ID)) {
+        if (seenChips[ID]->isErrorSet()) {
+          continue;
+        }
+        currentChip = *seenChips[ID];
+      }
+      AlpideCoder encoder;
+      encoder.encodeChip(reconstructedData, currentChip, localID,
+                         /*dummy bc*/ 0, currentChip.getROFlags());
+    }
+
+    // Pad the end of the reconstructed buffer with the zero bytes.
+    if (buffer.getSize() > reconstructedData.getSize())
+      reconstructedData.fill(0x00,
+                             buffer.getSize() - reconstructedData.getSize());
+
+    auto reportError = [&](std::string message) {
+      LOG(error) << "Error during decoder verification: " << message;
+      LOG(debug) << "Raw Data:";
+      buffer.rewind();
+      uint8_t dataC = 0;
+      int index = 1;
+      while (buffer.next(dataC)) {
+        LOG(debug) << index << ". 0x" << std::setfill('0') << std::setw(2)
+                   << std::hex << std::uppercase << (0xFF & dataC);
+        ++index;
+      }
+      LOG(debug) << "Reconstructed Data:";
+      reconstructedData.rewind();
+      index = 1;
+      while (reconstructedData.next(dataC)) {
+        LOG(debug) << index << ". 0x" << std::setfill('0') << std::setw(2)
+                   << std::hex << std::uppercase << (0xFF & dataC);
+        ++index;
+      }
+    };
+
+    // The reconstructed buffer is very similar to the original data flow
+    // with the exception to several pieces of information that get lost
+    // during the decoding:
+    // 1. Error trigger words: these are absent in the reconstructed buffer.
+    //    In case of BUSYON/BUSYOFF words, the verification is allowed to
+    //    continue.
+    // 2. Bunch counter for frame: the reconstructed buffer does contain the
+    //    corresponding words, but has a dummy value.
+
+    buffer.rewind();
+    while (true) {
+      uint8_t dataRec = 0;
+      uint8_t dataRaw = 0;
+
+      if (reconstructedData.isEmpty() || buffer.isEmpty()) {
+        // If either buffer is empty, verify that both buffers reached the end.
+        // If one of the streams is non-empty, then verify that the remaining
+        // bytes are zeroes.
+        if (reconstructedData.isEmpty() && buffer.isEmpty()) {
+          break;
+        }
+        PayLoadCont& nonEmptyBuffer =
+          !buffer.isEmpty() ? buffer : reconstructedData;
+        uint8_t dataC = 0;
+        while (nonEmptyBuffer.next(dataC)) {
+          if (dataC != 0x00) {
+            reportError("Buffer sizes mismatch.");
+            return false;
+          }
+        }
+        break;
+      }
+
+      reconstructedData.current(dataRec);
+      buffer.current(dataRaw);
+      if (dataRaw == dataRec) {
+        if ((dataRaw & (~MaskChipID)) == CHIPHEADER ||
+            (dataRaw & (~MaskChipID)) == CHIPEMPTY) {
+          // If the data correspond to the CHIPHEADER or CHIPEMPTY data words,
+          // skip the next byte that represent bunch counters.
+          buffer.next(dataRaw);
+          reconstructedData.next(dataRec);
+        }
+        buffer.next(dataRaw);
+        reconstructedData.next(dataRec);
+        continue;
+      }
+
+      if (dataRaw == BUSYON || dataRaw == BUSYOFF) {
+        // Placement of BUSYON and BUSYOFF triggers is arbitrary, just ignore
+        // the byte in the raw stream and move forward.
+        buffer.next(dataRaw);
+        continue;
+      }
+      if ((dataRaw & ERROR_MASK) == ERROR_MASK) {
+        // Placement of error triggers is arbitrary and we cannot guarantee
+        // the decoder to have processed the rest of the stream.
+        return false;
+      }
+
+      if ((dataRaw & (~MaskChipID)) == CHIPHEADER) {
+        // If we encounter a mismatch when the raw data is at the beginning
+        // of the next chip, it might indicate the fact that an error occured
+        // during the decoding of this chip. In this case, we cannot continue
+        // the byte stream verification after this point.
+        uint16_t ID = cidGetter(dataRaw & MaskChipID);
+        if (seenChips.count(ID) && seenChips[ID]->isErrorSet()) {
+          LOG(warning) << "Chip " << ID
+                       << " could not be verified due to set errors: "
+                       << ChipStat::reportErrors(*seenChips[ID]);
+          // TODO: Instead of returning, we can skip the part of the buffer
+          // that describes this chip and continue with the verification of
+          // the next one.
+          return true;
+        }
+      }
+
+      // If the read bytes is not related to the special cases, report error.
+      std::stringstream errorStream;
+      errorStream << "Unexpected byte mismatch during decoder verification. "
+                     "Expected: 0x"
+                  << std::hex << std::uppercase << std::setfill('0')
+                  << std::setw(2) << (0xFF & dataRaw) << ", Reconstructed: 0x"
+                  << std::setfill('0') << std::setw(2) << (0xFF & dataRec);
+      reportError(errorStream.str());
+      return false;
+    }
+    return true;
   }
 
   /// check if the byte corresponds to chip_header or chip_empty flag

--- a/Detectors/ITSMFT/common/reconstruction/include/ITSMFTReconstruction/ChipMappingITS.h
+++ b/Detectors/ITSMFT/common/reconstruction/include/ITSMFTReconstruction/ChipMappingITS.h
@@ -141,12 +141,52 @@ class ChipMappingITS
     }
   }
 
-  ///< get chip global SW ID from chipID on module, cable SW ID and stave (RU) info
-  uint16_t getGlobalChipID(uint16_t chOnModuleHW, int cableHW, const RUInfo& ruInfo) const
+  ///< get chipID on module from chip global SW ID, cable SW ID and stave (RU) info
+  uint16_t getLocalChipID(uint16_t globalID, int cableHW, const RUInfo& ruInfo)
+    const
   {
-    if (chOnModuleHW < MaxHWChipIDPerModuleSB[ruInfo.ruType] && cableHW <= MaxHWCableID[ruInfo.ruType]) {
-      uint16_t chipOnRU = ruInfo.ruType == IB ? (cableHW == chOnModuleHW ? cableHW : 0xff) : (ruInfo.ruType == MB ? HWCableHWChip2ChipOnRU_MB[cableHW][chOnModuleHW] : HWCableHWChip2ChipOnRU_OB[cableHW][chOnModuleHW]);
-      // uint16_t chipOnRU = ruInfo.firstChipIDSW + mCableHWFirstChip[ruInfo.ruType][cableHW] + chipModuleIDHW2SW(ruInfo.ruType, chOnModuleHW);  // This is an old way, w/o error check
+    if (cableHW <= MaxHWCableID[ruInfo.ruType] && globalID != 0xffff) {
+      uint16_t chipOnRU = globalID - ruInfo.firstChipIDSW;
+      switch (ruInfo.ruType) {
+        case IB:
+          return chipOnRU;
+        case MB:
+          for (int ihw = 0; ihw < 15; ihw++) {
+            if (HWCableHWChip2ChipOnRU_MB[cableHW][ihw] == chipOnRU)
+              return ihw;
+          }
+          break;
+        case OB:
+          for (int ihw = 0; ihw < 15; ihw++) {
+            if (HWCableHWChip2ChipOnRU_OB[cableHW][ihw] == chipOnRU)
+              return ihw;
+          }
+          break;
+      }
+    }
+    return 0xffff;
+  }
+
+  ///< get chip global SW ID from chipID on module, cable SW ID and stave (RU)
+  /// info
+  uint16_t getGlobalChipID(uint16_t chOnModuleHW, int cableHW,
+                           const RUInfo& ruInfo) const
+  {
+    if (chOnModuleHW < MaxHWChipIDPerModuleSB[ruInfo.ruType] &&
+        cableHW <= MaxHWCableID[ruInfo.ruType]) {
+      uint16_t chipOnRU = 0xff;
+      switch (ruInfo.ruType) {
+        case IB:
+          if (cableHW == chOnModuleHW)
+            chipOnRU = cableHW;
+          break;
+        case MB:
+          chipOnRU = HWCableHWChip2ChipOnRU_MB[cableHW][chOnModuleHW];
+          break;
+        case OB:
+          chipOnRU = HWCableHWChip2ChipOnRU_OB[cableHW][chOnModuleHW];
+          break;
+      }
       return chipOnRU < 0xff ? ruInfo.firstChipIDSW + chipOnRU : 0xffff;
     }
     return 0xffff;

--- a/Detectors/ITSMFT/common/reconstruction/include/ITSMFTReconstruction/ChipMappingMFT.h
+++ b/Detectors/ITSMFT/common/reconstruction/include/ITSMFTReconstruction/ChipMappingMFT.h
@@ -132,6 +132,13 @@ class ChipMappingMFT
   ///< convert cable iterator ID to its position on the ActiveLanes word in the GBT.header for given RU type
   uint8_t cablePos(uint8_t ruType, uint8_t id) const { return mCablePos[ruType][id]; }
 
+  ///< get chipID on module from chip global SW ID, cable SW ID and stave (RU) info
+  uint16_t getLocalChipID(uint16_t globalID, int cableHW, const RUInfo& ruInfo) const
+  {
+    /// TODO
+    return 0xffff;
+  }
+
   ///< get chip global SW ID from chipID on module, cable SW ID and stave (RU) info
   uint16_t getGlobalChipID(uint16_t chOnModuleHW, int cableHW, const RUInfo& ruInfo) const
   {

--- a/Detectors/ITSMFT/common/reconstruction/include/ITSMFTReconstruction/DecodingStat.h
+++ b/Detectors/ITSMFT/common/reconstruction/include/ITSMFTReconstruction/DecodingStat.h
@@ -177,6 +177,7 @@ struct ChipStat {
   uint32_t getNErrors() const;
   uint32_t addErrors(uint32_t mask, uint16_t chID, int verbosity);
   uint32_t addErrors(const ChipPixelData& d, int verbosity);
+  static std::string reportErrors(const ChipPixelData& d);
   void print(bool skipNoErr = true, const std::string& pref = "FEEID") const;
 
   ClassDefNV(ChipStat, 1);

--- a/Detectors/ITSMFT/common/reconstruction/include/ITSMFTReconstruction/RUDecodeData.h
+++ b/Detectors/ITSMFT/common/reconstruction/include/ITSMFTReconstruction/RUDecodeData.h
@@ -36,6 +36,8 @@ struct RUDecodeData {
 
   std::array<PayLoadCont, MaxCablesPerRU> cableData{};                        // cable data in compressed ALPIDE format
   std::vector<o2::itsmft::ChipPixelData> chipsData{};                         // fully decoded data in 1st nChipsFired chips
+  std::map<int, ChipPixelData*> seenChips{};                                  // Map that records chip data by global ID
+  std::vector<uint16_t> seenChipIDsInCable{};                                 // IDs of chips seen in the current cable
   std::vector<uint16_t> seenChipIDs{};                                        // IDs of all chips seen during ROF decoding, including empty ones
   std::array<int, MaxLinksPerRU> links{};                                     // link entry RSTODO: consider removing this and using pointer
   std::array<uint8_t, MaxCablesPerRU> cableHWID{};                            // HW ID of cable whose data is in the corresponding slot of cableData
@@ -65,7 +67,7 @@ struct RUDecodeData {
   }
   void setROFInfo(ChipPixelData* chipData, const GBTLink* lnk);
   template <class Mapping>
-  int decodeROF(const Mapping& mp, const o2::InteractionRecord ir);
+  int decodeROF(const Mapping& mp, const o2::InteractionRecord ir, bool verifyDecoder);
   void fillChipStatistics(int icab, const ChipPixelData* chipData);
   void dumpcabledata(int icab);
   bool checkLinkInSync(int icab, const o2::InteractionRecord ir);
@@ -75,7 +77,7 @@ struct RUDecodeData {
 ///_________________________________________________________________
 /// decode single readout frame, the cable's data must be filled in advance via GBTLink::collectROFCableData
 template <class Mapping>
-int RUDecodeData::decodeROF(const Mapping& mp, const o2::InteractionRecord ir)
+int RUDecodeData::decodeROF(const Mapping& mp, const o2::InteractionRecord ir, bool verifyDecoder)
 {
   nChipsFired = 0;
   lastChipChecked = 0;
@@ -98,11 +100,25 @@ int RUDecodeData::decodeROF(const Mapping& mp, const o2::InteractionRecord ir)
       auto chip = mp.getGlobalChipID(cid, cabHW, *this->ruInfo);
       return chip;
     };
+    auto localChipIdGetter = [this, &mp, cabHW](int gid) {
+      auto localID = mp.getLocalChipID(gid, cabHW, *this->ruInfo);
+      return localID;
+    };
+
     int ret = 0;
     // dumpcabledata(icab);
 
-    while ((ret = AlpideCoder::decodeChip(*chipData, cableData[icab], seenChipIDs, chIdGetter)) || chipData->isErrorSet()) { // we register only chips with hits or errors flags set
+    std::vector<uint16_t>* seenChipIDsPtr = &seenChipIDs;
+    if (verifyDecoder) {
+      seenChipIDsInCable.clear();
+      seenChipIDsPtr = &seenChipIDsInCable;
+      seenChips.clear();
+    }
+    while ((ret = AlpideCoder::decodeChip(*chipData, cableData[icab], *seenChipIDsPtr, chIdGetter)) || chipData->isErrorSet()) { // we register only chips with hits or errors flags set
       setROFInfo(chipData, cableLinkPtr[icab]);
+      if (verifyDecoder) {
+        seenChips[chipData->getChipID()] = chipData;
+      }
       auto nhits = chipData->getData().size();
       if (nhits && doneChips[chipData->getChipID()]) {
         if (chipData->getChipID() == chipsData[nChipsFired - 1].getChipID()) {
@@ -131,6 +147,12 @@ int RUDecodeData::decodeROF(const Mapping& mp, const o2::InteractionRecord ir)
       if (ret < 0) {
         break; // negative code was returned by decoder: abandon cable data
       }
+    }
+    if (verifyDecoder) {
+      AlpideCoder::verifyDecodedCable(seenChips, cableData[icab],
+                                      seenChipIDsInCable, localChipIdGetter,
+                                      chIdGetter);
+      seenChipIDs.insert(seenChipIDs.end(), seenChipIDsInCable.begin(), seenChipIDsInCable.end());
     }
     cableData[icab].clear();
   }

--- a/Detectors/ITSMFT/common/reconstruction/include/ITSMFTReconstruction/RawPixelDecoder.h
+++ b/Detectors/ITSMFT/common/reconstruction/include/ITSMFTReconstruction/RawPixelDecoder.h
@@ -107,6 +107,9 @@ class RawPixelDecoder final : public PixelReader
   void setAllowEmptyROFs(bool v) { mAlloEmptyROFs = v; }
   bool getAllowEmptyROFs() const { return mAlloEmptyROFs; }
 
+  void setVerifyDecoder(bool v) { mVerifyDecoder = v; }
+  bool getVerifyDecoder() const { return mVerifyDecoder; }
+
   void setInstanceID(size_t i) { mInstanceID = i; }
   void setNInstances(size_t n) { mNInstances = n; }
   auto getInstanceID() const { return mInstanceID; }
@@ -160,6 +163,7 @@ class RawPixelDecoder final : public PixelReader
   bool mAlloEmptyROFs = false;                                                        // do not skip empty ROFs
   bool mROFRampUpStage = false;                                                       // are we still in the ROF ramp up stage?
   bool mSkipRampUpData = false;
+  bool mVerifyDecoder = false;
   int mVerbosity = 0;
   int mNThreads = 1; // number of decoding threads
   // statistics

--- a/Detectors/ITSMFT/common/reconstruction/src/DecodingStat.cxx
+++ b/Detectors/ITSMFT/common/reconstruction/src/DecodingStat.cxx
@@ -92,6 +92,20 @@ void ChipStat::print(bool skipNoErr, const std::string& pref) const
 }
 
 ///_________________________________________________________________
+/// print chip decoding statistics
+std::string ChipStat::reportErrors(const ChipPixelData& d)
+{
+  std::string res;
+  for (int i = NErrorsDefined; i--;) {
+    if (d.getErrorFlags() & (0x1UL << i)) {
+      res += ErrNames[i];
+      res += " ";
+    }
+  }
+  return res;
+}
+
+///_________________________________________________________________
 /// print link decoding statistics
 void GBTLinkDecodingStat::print(bool skipNoErr) const
 {

--- a/Detectors/ITSMFT/common/reconstruction/src/RawPixelDecoder.cxx
+++ b/Detectors/ITSMFT/common/reconstruction/src/RawPixelDecoder.cxx
@@ -109,7 +109,7 @@ int RawPixelDecoder<Mapping>::decodeNextTrigger()
       auto& ru = mRUDecodeVec[iru];
       if (ru.nNonEmptyLinks) {
         ru.ROFRampUpStage = mROFRampUpStage;
-        mNPixelsFiredROF += ru.decodeROF(mMAP, mInteractionRecord);
+        mNPixelsFiredROF += ru.decodeROF(mMAP, mInteractionRecord, mVerifyDecoder);
         mNChipsFiredROF += ru.nChipsFired;
       } else {
         ru.clearSeenChipIDs();

--- a/Detectors/ITSMFT/common/workflow/include/ITSMFTWorkflow/STFDecoderSpec.h
+++ b/Detectors/ITSMFT/common/workflow/include/ITSMFTWorkflow/STFDecoderSpec.h
@@ -46,6 +46,7 @@ struct STFDecoderInp {
   bool doSquashing = false;
   bool askSTFDist = true;
   bool allowReporting = true;
+  bool verifyDecoder = false;
   o2::header::DataOrigin origin{"NIL"};
   std::string deviceName{};
   std::string inputSpec{};
@@ -79,6 +80,7 @@ class STFDecoder : public Task
   bool mAllowReporting = true;
   bool mApplyNoiseMap = true;
   bool mUseClusterDictionary = true;
+  bool mVerifyDecoder = false;
   int mDumpOnError = 0;
   int mNThreads = 1;
   int mVerbosity = 0;

--- a/Detectors/ITSMFT/common/workflow/src/STFDecoderSpec.cxx
+++ b/Detectors/ITSMFT/common/workflow/src/STFDecoderSpec.cxx
@@ -47,7 +47,7 @@ using namespace o2::framework;
 ///_______________________________________
 template <class Mapping>
 STFDecoder<Mapping>::STFDecoder(const STFDecoderInp& inp, std::shared_ptr<o2::base::GRPGeomRequest> gr)
-  : mDoClusters(inp.doClusters), mDoPatterns(inp.doPatterns), mDoDigits(inp.doDigits), mDoCalibData(inp.doCalib), mAllowReporting(inp.allowReporting), mInputSpec(inp.inputSpec), mGGCCDBRequest(gr)
+  : mDoClusters(inp.doClusters), mDoPatterns(inp.doPatterns), mDoDigits(inp.doDigits), mDoCalibData(inp.doCalib), mAllowReporting(inp.allowReporting), mVerifyDecoder(inp.verifyDecoder), mInputSpec(inp.inputSpec), mGGCCDBRequest(gr)
 {
   mSelfName = o2::utils::Str::concat_string(Mapping::getName(), "STFDecoder");
   mTimer.Stop();
@@ -98,6 +98,7 @@ void STFDecoder<Mapping>::init(InitContext& ic)
     mDecoder->setAllowEmptyROFs(ic.options().get<bool>("allow-empty-rofs"));
     mDecoder->setRawDumpDirectory(dumpDir);
     mDecoder->setFillCalibData(mDoCalibData);
+    mDecoder->setVerifyDecoder(mVerifyDecoder);
     bool ignoreRampUp = !ic.options().get<bool>("accept-rof-rampup-data");
     mDecoder->setSkipRampUpData(ignoreRampUp);
   } catch (const std::exception& e) {

--- a/Detectors/ITSMFT/common/workflow/src/stf-decoder-workflow.cxx
+++ b/Detectors/ITSMFT/common/workflow/src/stf-decoder-workflow.cxx
@@ -22,6 +22,7 @@ void customize(std::vector<o2::framework::ConfigParamSpec>& workflowOptions)
 {
   // option allowing to set parameters
   std::vector<ConfigParamSpec> options{
+    ConfigParamSpec{"verify", VariantType::Bool, false, {"verify ALPIDE decoder by re-encoding the data"}},
     ConfigParamSpec{"runmft", VariantType::Bool, false, {"source detector is MFT (default ITS)"}},
     ConfigParamSpec{"no-clusters", VariantType::Bool, false, {"do not produce clusters (def: produce)"}},
     ConfigParamSpec{"no-cluster-patterns", VariantType::Bool, false, {"do not produce clusters patterns (def: produce)"}},
@@ -50,6 +51,7 @@ WorkflowSpec defineDataProcessing(ConfigContext const& cfgc)
   inp.doCalib = cfgc.options().get<bool>("enable-calib-data");
   inp.doSquashing = cfgc.options().get<bool>("squash-overflow-pixels");
   inp.askSTFDist = !cfgc.options().get<bool>("ignore-dist-stf");
+  inp.verifyDecoder = cfgc.options().get<bool>("verify");
   inp.inputSpec = cfgc.options().get<std::string>("dataspec");
   // Update the (declared) parameters if changed from the command line
   o2::conf::ConfigurableParam::updateFromString(cfgc.options().get<std::string>("configKeyValues"));

